corehq/apps/api/user_updates.py

```diff
@@ -165,3 +167,53 @@ def _log_phone_number_change(new_phone_numbers, old_phone_numbers, user_change_l
 
     if change_messages:
         user_change_logger.add_change_message({'phone_numbers': change_messages})
+
+
+def _update_location(user, location_object, user_change_logger):
+    primary_location_id = location_object.get('primary_location')
+    location_ids = location_object.get('locations')
+
+    if primary_location_id is None and location_ids is None:
+        return
+
+    user_current_primary_location_id = user.location_id
+
+    if not primary_location_id and not location_ids:
+        # we will have bundle.obj.save() later, so when change locations, commit=False
+        user.unset_location(commit=False)
+        user.reset_locations([], commit=False)
+        if user_change_logger:
+            user_change_logger.add_changes({'location_id': primary_location_id})
+            user_change_logger.add_info(UserChangeMessage.primary_location_removed())
+            user_change_logger.add_changes({'assigned_location_ids': location_ids})
+            user_change_logger.add_info(UserChangeMessage.assigned_locations_info([]))
+    elif not primary_location_id or not location_ids:
+        raise UpdateUserException(_('Both primary_location and locations must be provided together.'))
```

Comment:
I know we discussed this kind of thing a bunch - this approach seems appropriate to me, good to default to the more explicit option, especially in an API